### PR TITLE
RFC on joiner

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

File: go/libraries/doltcore/schema/schema_impl.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func SchemaFromCols(allCols *ColCollection) Schema {
 	}
 
 	if len(pkCols) == 0 {
-		panic("no primary key columns specified")
```
Comment (bheni) [inline go/libraries/doltcore/schema/schema_impl.go]: should this be legal?

Comment (bheni) [reply go/libraries/doltcore/schema/schema_impl.go]: Maybe we should enforce pks on schemas in a different place.  Such as when you attempt to create a table with the schema.

Comment (bheni) [reply go/libraries/doltcore/schema/schema_impl.go]: I think we must enforce it in other places already, otherwise we would be seeing a bunch of panics.

Comment (zachmu) [reply go/libraries/doltcore/schema/schema_impl.go]: We have UnkeyedSchemaFromColsfor this, which doesn't enforce the primary key check. We use that for e.g. SQL result sets and other unkeyed schemas.

**Pull request closed without merging**